packit/upstream.py

```diff
@@ -433,9 +433,13 @@ def fix_spec(self, archive: str, version: str, commit: str):
             g_desc_raw = git_des_out.rsplit("-", 2)[1:]
             # release components are meant to be separated by ".", not "-"
             git_desc_suffix = "." + ".".join(g_desc_raw)
+        try:
+            current_branch = run_command(["git", "rev-parse", "--abbrev-ref", "HEAD"], output=True).strip()
```

Comment:
Why not use `self._local_project.ref` here? Wouldn't that provide the same information?

Reply:
Yes, `self._local_project.ref` returns commit hash if we are on a detached HEAD and branch name otherwise.

Reply:
And btw, I would use `self.local_project` instead of `self._local_project` to support lazy instantiation.